src/ast/passes/codegen_llvm.cpp

```diff
@@ -2635,6 +2634,37 @@ void CodegenLLVM::visit(AttachPoint &)
   // Empty
 }
 
+// For older 5.2 kernels, we need to bound CPU ID by MAX_CPU_ID to pass
+// BPF verifier on older kernels.
+//
+// However, for programs that do many scratch buffer lookups, this causes
+// many jumps which means a complex program may be rejected by the BPF
+// verifier for >= 8192 jumps (current limit).
+//
+// To minimize jumps, we bound the CPU ID at the beginning of each function
+// and cache the value in a register. This ensures we only add one
+// additional jump per function. Note we cannot cache this value in a global
+// variable since memory is shared between all CPUs or a per-CPU array since
+// that would be recursive.
+//
+// Note the only functions in scope are the ones where scratch buffer is used
+// which is currently subprograms, probes and map_for_each_cb. To account for
+// nesting, we use a stack to ensure we're referencing the correct instance.
+Value *CodegenLLVM::generateCachedCpuId(const location &loc)
+{
+  if (!bpftrace_.resources.needed_global_vars.count(
+          globalvars::GlobalVar::MAX_CPU_ID)) {
+    return nullptr;
+  }
+
+  auto cpu_id = b_.CreateGetCpuId(loc);
+  auto max = b_.CreateLoad(b_.getInt64Ty(),
+                           module_->getGlobalVariable(to_string(
+                               bpftrace::globalvars::GlobalVar::MAX_CPU_ID)));
+  auto cmp = b_.CreateICmp(CmpInst::ICMP_ULE, cpu_id, max);
+  return b_.CreateSelect(cmp, cpu_id, max);
```

Comment:
Why get rid of the descriptive names that were used for these in IRBuilder? They make the generated IR a lot easier to read (and it seems that 99% of the expected results changed in this PR are due to the names disappearing)

Reply:
@ajor the reason I got rid of this is when I hit the a bug re: for loops that looked like `Instruction does not dominate all use cases`: https://www.reddit.com/r/LLVM/comments/vd5urh/runtimeerror_instruction_does_not_dominate_all/ (it was due to not using stack + for loop)

With named variables, it was very hard to debug because all the names looked the same. When I removed the names, it was obvious that function call 1 was trying to use a register from function call 2.

I will add them back in another PR, sorry for the confusion